src/source/Ice/IceAgentStateMachine.c

```diff
@@ -886,6 +886,11 @@ STATUS executeReadyIceAgentState(UINT64 customData, UINT64 time)
 
     pIceAgent->iceAgentState = ICE_AGENT_STATE_READY;
 
+    if (!IS_CANN_PAIR_SENDING_FROM_RELAYED(pIceAgent->pDataSendingIceCandidatePair) && pIceAgent->pTurnConnection != NULL) {
```

Comment:
`IS_CANN_PAIR_SENDING_FROM_RELAYED` just checks if the candidate pair is relay connection or not, does it mean we are going to stop the connection for all the relay connection, i.e. turn connection?

Reply:
IS_CANN_PAIR_SENDING_FROM_RELAYED checks whether the local candidate of the selected pair is relay or not. if not, it means we are not using turn, thus we can safely shutdown turnConnection.